core/templates/components/forms/custom-forms-directives/thumbnail-uploader.component.ts

```diff
@@ -34,37 +34,37 @@ import { EditThumbnailModalComponent } from './edit-thumbnail-modal.component';
   templateUrl: './thumbnail-uploader.component.html'
 })
 export class ThumbnailUploaderComponent implements OnInit, OnChanges {
-  @Input() disabled: boolean;
-  @Input() useLocalStorage: boolean;
-  @Input() allowedBgColors: string[];
-  @Input() aspectRatio: string;
-  @Input() bgColor: string;
-  @Input() filename: string;
-  @Input() previewDescription: string;
-  @Input() previewDescriptionBgColor: string;
-  @Input() previewFooter: string;
-  @Input() previewTitle: string;
-  @Output() updateBgColor: EventEmitter<string> = new EventEmitter() ;
+  @Input() disabled!: boolean;
+  @Input() useLocalStorage!: boolean;
+  @Input() allowedBgColors!: string[];
+  @Input() aspectRatio!: string;
+  @Input() bgColor!: string;
+  @Input() filename!: string;
+  @Input() previewDescription!: string;
+  @Input() previewDescriptionBgColor!: string;
+  @Input() previewFooter!: string;
+  @Input() previewTitle!: string;
+  @Output() updateBgColor: EventEmitter<string> = new EventEmitter();
   @Output() updateFilename: EventEmitter<string> = new EventEmitter();
   @Output() imageSave: EventEmitter<void> = new EventEmitter();
-  openInUploadMode: boolean;
-  tempBgColor: string;
-  tempImageName: string;
-  uploadedImage: string;
-  uploadedImageMimeType: string;
-  dimensions: { height: number; width: number };
-  resampledFile: Blob;
-  newThumbnailDataUrl: string;
-  localStorageBgcolor: string;
-  imageUploadUrlTemplate: string;
+  openInUploadMode: boolean | undefined;
+  tempBgColor: string | undefined;
+  tempImageName: string | undefined;
+  uploadedImage: string | undefined;
+  uploadedImageMimeType: string | undefined;
+  dimensions: { height: number; width: number } | undefined;
+  resampledFile: Blob | undefined;
+  newThumbnailDataUrl: string | undefined;
+  localStorageBgcolor: string | undefined;
+  imageUploadUrlTemplate: string | undefined;
```

Comment:
Why are all of these `undefined`?

Reply:
These were not definitely assigned in the constructor. Since these fields may be undefined, it should be marked in the field declaration either by using the ```?``` or by typing the field as ```type |  undefined```

![image](https://user-images.githubusercontent.com/81933624/147964362-44e3cd86-e381-4b87-a24c-218444d3fd3d.png)

Reply:
And do we need to define all of these here? If yes then it needs to be properly documented.